### PR TITLE
[CI] Add m6g instance (ARM64) to CI

### PR DESCRIPTION
**Pull request by zhiics**

### CODE REVIEW

File: docker/Dockerfile.ci_arm [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+RUN apt-get update --fix-missing
+RUN apt-get install -y ca-certificates
+RUN apt-get install -y gnupg2
+
+COPY install/ubuntu_install_core.sh /install/ubuntu_install_core.sh
+RUN bash /install/ubuntu_install_core.sh
+
+COPY install/ubuntu_install_llvm.sh /install/ubuntu_install_llvm.sh
+RUN bash /install/ubuntu_install_llvm.sh
```
Comment (leandron) [inline docker/Dockerfile.ci_arm:31]: This script uses `apt.llvm.org` repository, which does not publish AArch64 (arm64) builds. So, if you want to use the same LLVM versions we have for other builds, for now, one option would be downloading and uncompressing an LLVM build from the official source:
- LLVM 11: https://github.com/llvm/llvm-project/releases/download/llvmorg-11.0.0/clang+llvm-11.0.0-aarch64-linux-gnu.tar.xz
- LLVM 10: https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/clang+llvm-10.0.0-aarch64-linux-gnu.tar.xz
- LLVM 9: https://releases.llvm.org/9.0.0/clang+llvm-9.0.0-aarch64-linux-gnu.tar.xz

Once arm64 builds are published into apt.llvm.org, then we can just share the same existing scripts.

File: docker/Dockerfile.ci_arm [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# CI docker arm env
+# tag: v0.70
```
Comment (leandron) [inline docker/Dockerfile.ci_arm:19]: minor: just for completeness, maybe this should match with the tag you want, or be removed if not relevant?

File: docker/Dockerfile.ci_arm [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+FROM ubuntu:18.04
+
+RUN apt-get update --fix-missing
+RUN apt-get install -y ca-certificates
+RUN apt-get install -y gnupg2
```
Comment (leandron) [inline docker/Dockerfile.ci_arm:25]: This is likely to increase the space taken by the resultant Docker image. It is recommended ([by Docker](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run)) to line-up apt-get calls in one statement, and make sure you remove the apt-cache by the end.

**Review by leandron — COMMENTED** [review]
Thanks for this PR! It looks like a good improvement for our upstream CI.

A few comments based on some experiments I did in that direction, in past.

File: docker/Dockerfile.ci_arm (hunk 1 above)
Comment (zhiics) [reply docker/Dockerfile.ci_arm:31]: The current one is more like experimental. I did test it locally. It seems every was okay. I will probably leave these arm specific installation (including acl etc) to your folks, how do you think?

**Pull request merged**